From 79f49287b488384f5577d3b759710ce56533e5ed Mon Sep 17 00:00:00 2001
From: "xuesong.jiang" <xuesong.jiang@amlogic.com>
Date: Wed, 17 May 2023 09:18:48 +0000
Subject: [PATCH 3/3] amlv4l2dec: CB1 deal invalid pts [1/1]

PD#SWPL-113634

Problem:
Identify invalid pts obtained from v4l2 and convert them into gst invalid pts

Solution:
Identify invalid pts obtained from v4l2 and convert them into gst invalid pts

Verify:
(detail info)

Change-Id: I9cf995855a73837aaf0867b98316f51b2351f8e7
Signed-off-by: xuesong.jiang <xuesong.jiang@amlogic.com>
---
 src/gstamlv4l2bufferpool.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/gstamlv4l2bufferpool.c b/src/gstamlv4l2bufferpool.c
index dcd6c7d..838f3fc 100644
--- a/src/gstamlv4l2bufferpool.c
+++ b/src/gstamlv4l2bufferpool.c
@@ -1422,7 +1422,10 @@ gst_aml_v4l2_buffer_pool_dqbuf(GstAmlV4l2BufferPool *pool, GstBuffer **buffer,
         GST_OBJECT_UNLOCK(pool);
     }
 
-    timestamp = GST_TIMEVAL_TO_TIME(group->buffer.timestamp);
+    if (-1== group->buffer.timestamp.tv_sec)
+        timestamp = GST_CLOCK_TIME_NONE;
+    else
+        timestamp = GST_TIMEVAL_TO_TIME(group->buffer.timestamp);
 
     size = 0;
     vmeta = gst_buffer_get_video_meta(outbuf);
-- 
2.25.1

